extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -1184,22 +1116,25 @@ private static void verifyFileLayout(
      *
      * @param directoryUri the URI for the root directory to search for .parquet files
      * @param readInstructions the instructions for customizations while reading
+     * @param channelsProvider the provider for creating seekable channels. If null, a new provider will be created and
+     *        used for all channels created while reading the table.
      * @return the table
      */
     private static Table readKeyValuePartitionedTable(
             @NotNull final URI directoryUri,
-            @NotNull final ParquetInstructions readInstructions) {
+            @NotNull final ParquetInstructions readInstructions,
+            @Nullable final SeekableChannelsProvider channelsProvider) {
         verifyFileLayout(readInstructions, ParquetFileLayout.KV_PARTITIONED);
         if (readInstructions.getTableDefinition().isEmpty()) {
-            return readTable(new ParquetKeyValuePartitionedLayout(directoryUri,
-                    MAX_PARTITIONING_LEVELS_INFERENCE, readInstructions), readInstructions);
+            return readTable(ParquetKeyValuePartitionedLayout.create(directoryUri,
+                    MAX_PARTITIONING_LEVELS_INFERENCE, readInstructions, channelsProvider), readInstructions);
         }
         final TableDefinition tableDefinition = readInstructions.getTableDefinition().get();
         if (tableDefinition.getColumnStream().noneMatch(ColumnDefinition::isPartitioning)) {
             throw new IllegalArgumentException("No partitioning columns");
         }
```

Comment:
I think this check is invalid now. If the user specifies a table def that matches a flat layout, but does not explicitly specify a flat layout, we'll blow up.